pkg/operator/configobservation/etcdendpoints/observe_etcd_endpoints.go

```diff
@@ -0,0 +1,110 @@
+package etcdendpoints
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation"
+	endpointsobserver "github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation/etcdobserver"
+)
+
+const (
+	etcdEndpointNamespace = "openshift-etcd"
+	etcdEndpointName      = "etcd-endpoints"
+)
+
+// BackwardsCompatibleObserveStorageURLs will prefer to observe the new
+// etcd-endpoints configmap, and will fall back to reading the endpoint
+// resource only if the configmap doesn't yet exist.
+func BackwardsCompatibleObserveStorageURLs(genericListers configobserver.Listers, recorder events.Recorder, currentConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	if _, err := listers.ConfigmapLister.ConfigMaps(etcdEndpointNamespace).Get(etcdEndpointName); err == nil {
+		return ObserveStorageURLs(genericListers, recorder, currentConfig)
+	} else if !errors.IsNotFound(err) {
+		return nil, []error{err}
+	}
+	recorder.Warningf("ObserveStorageFallback", "%s/%s configmap not found, falling back to host-etcd-2 endpoint", etcdEndpointNamespace, etcdEndpointName)
```

Comment:
@sanchezl @deads2k this does generate a lot of event noise before the corresponding etcd-operator transition (3212 events in the e2e-aws run), but that should of course taper off and cease entirely when etcd-operator is updated and generates the configmap.

I'm not sure if the event volume is concerning on its own, or whether the warning provides enough value to justify it. I'll keep it for now, let me know what you think.

Reply:
> I'm not sure if the event volume is concerning on its own, or whether the warning provides enough value to justify it. I'll keep it for now, let me know what you think.

it does not.  remove it please

Reply:
Fixed